### ISSUE TITLE
Comparison  function is error for struct host_t

### ISSUE THREAD

**Issue by MatthewButterfly** [luca3m/redis3m#50]
Below code in cluster_pool.h is to define less operator for host_t. The class host_t is used as key in a map.  
I think the implemetion has a error.
Below two redis host are different address, they should be as different values and can be inserted into a map. But the current implemention will treat them as the same value ,since the port are the same, the less function will return false. 
Please have a look at it.
redis1 :  address: *192.168.10.10*, port: 6379
redis2 : address: *192.168.10.11*, prot: 6379

```
inline bool operator < (const host_t& rvalue) const
{
      return (address <= rvalue.address) && ( port < rvalue.port);
}
```

**Comment by luca3m** [comment]
You're right, by the way the `cluster` implementation is pretty experimental and uncomplete at the moment.
